### PR TITLE
measly attempt at a ktx tool

### PR DESCRIPTION
**Pull request by cmeury**
Not sure what to really display in the preview window. What do you think?

screen shot 2018-03-24 at 23 13 21

(Note to future self: In the off-chance this PR is successful, need to make a PR for the [tap](https://github.com/blendle/homebrew-blendle/blob/master/Formula/kns.rb) as well. Also see help for [distributing shell scripts via homebrew](https://kylebebak.github.io/post/distribute-program-via-homebrew).)

### CODE REVIEW

File: helpers/ktx-preview.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+IFS=';' read -ra context <<< "$(kubectl config view -o jsonpath="{range .contexts[?(@.name == '${selected}')].context}{@.cluster};{@.user};{@.namespace};{end}")"
+cluster=${context[0]}
+user=${context[1]}
+namespace=${context[2]}
+
+server=$(kubectl config view -o jsonpath="{.clusters[?(@.name == '${cluster}')].cluster.server}")
+
+echo "=== ${selected} ===\n"
+echo "Cluster ${cluster};Server ${server};User ${user};Namespace ${namespace};" | tr ';' "\n" | column -t
```
Comment (koenbollen) [inline helpers/ktx-preview.sh]: Make sure to add a newline at the end of the file.

**Review by koenbollen — COMMENTED** [review]
Hello Cedric,
Neat addition. I like it.

Some quick notes: Try running [shellcheck](https://github.com/koalaman/shellcheck) on your code. It hints on some possible problems. ✅ 

Cheers!

File: helpers/ktx-preview.sh (hunk 1 above)
Comment (cmeury) [reply helpers/ktx-preview.sh]: Sure, will do. May I ask why?

Comment (koenbollen) [reply helpers/ktx-preview.sh]: It's part legacy and part ease of use for other developers (using bash for example).
I think these articles best describe it:
http://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/
https://robots.thoughtbot.com/no-newline-at-end-of-file

**Comment by cmeury** [comment]
Thanks @koenbollen. I pushed fixes for the shellcheck reports.

File: helpers/ktx-preview.sh (hunk 1 above)
Comment (cmeury) [reply helpers/ktx-preview.sh]: Today I learnt something new! Thanks for that, this somehow slipped my attention for all these years. Cool.

**Comment by cmeury** [comment]
So you think `ktx` is a thing? If yes, let me know, and I'll extend the README etc. as well.

**Comment by koenbollen** [comment]
Yeah, lets do this. I do however want to keep the focus on the `kns` tool where the ktx is an extra (my guess it that namespace switching is more common then context switching).

**Comment by cmeury** [comment]
Yes, I totally agree. `kns` is the star here. By the way, I was incredibly surprised to see how short the script is after using it for quite a while (**a lot**). Excellent job!

**Comment by cmeury** [comment]
@koenbollen Added a small `README` section.

**Comment by koenbollen** [comment]
Nice.
But shouldn't we just install ktx in the homebrew formula?

**Comment by cmeury** [comment]
Sure! Honestly, whatever you think is best. No strong opinions on my side. Should I remove the section then? Or what would be your suggestion?

**Comment by koenbollen** [comment]
Keep the section. Just explain that `ktx` is a part of this package and what it does. 👍

**Comment by cmeury** [comment]
@koenbollen Done

**Review by koenbollen — APPROVED** [review]

**Comment by koenbollen** [comment]
@cmeury I guess it's done. I'll try and get a colleague to check it out and then I'll merge it.

**Review by koenrh — APPROVED** [review]
Looks good to me. Thanks @cmeury!

@koenbollen Could you update the Homebrew formula to install both `kns` and `ktx`?

**Comment by koenbollen** [comment]
@koenrh 
> Could you update the Homebrew formula to install both kns and ktx?

I will! 👍

**Comment by koenbollen** [comment]
@cmeury 
I've updated our Homebrew tab: https://github.com/blendle/homebrew-blendle/pull/22

Thanks for your contribution! 🎉

**Comment by cmeury** [comment]
Awesome :) Happy switching!

**Pull request merged**